web/src/pages/Main.tsx

```diff
@@ -1,14 +1,65 @@
 import { Header } from '../components/common/Header'
-import { useContext } from 'react'
-import { AuthContext } from 'react-oauth2-code-pkce'
+import { useEffect, useState } from 'react'
+import { MoleTable } from '../components/feature/MoleTable'
+import MercuryAPI from '../api/MercuryAPI'
+import { AxiosResponse } from 'axios'
+
+// TODO: Get type from generated API
+export type TComponentResponse = {
+  [key: string]: { [key: string]: string }
+}
 
 export const MainPage = (): JSX.Element => {
-  const { token } = useContext(AuthContext)
+  const [components, setComponents] = useState<TComponentResponse>()
+  const [loading, setLoading] = useState<boolean>(false)
+
+  // Fetch list of components name once on page load
+  useEffect(() => {
+    const mercuryApi = new MercuryAPI('')
```

Comment:
This violates the Dependency Inversion Principle. Main will now create a new instance of the API on every render.. 
I suggest we make a custom hook for this instead.

Reply:
Bad change from splitting the original PR. Fixed

Reply:
You can move mercuryApi outside the useEffect, to prevent unnecessary initialisations. You can also wrap the fetch in an expression to check if components has a state, to prevent re-fetching.

Reply:
Moved api init to one level up.
The useEffect has an empty dependency array, and will only run once, when components has no state.